### DISCUSSION TITLE
Support for xparse?

### DISCUSSION THREAD

**Discussion by GregVernon** [KaTeX/KaTeX#3724] [Q&A] [answered]
We recently switched from using "pure" LaTeX to MathJax -- so that we can render our [theory manuals](https://docs.coreform.com/latest/html/coreform-public/iga-theory-frm.html#%28part._.Push_forward_to_current_configuration%29) / reports on our webpage _and also_ render to PDF (for publication / report submission etc). One issue that we're running into is that we made _heavy_ use of `xparse` and MathJax doesn't support `xparse`.  I was wondering if KaTeX supports `xparse`?  From [this discussion](https://github.com/KaTeX/KaTeX/discussions/3520) it sounds like "no" - but I didn't see it explicitly stated and am not expert enough to read between the lines whether it is or isn't supported.

If it's not supported, is there any advantage to using KaTeX vs MathJax _specifically_ regarding "workarounds" for lack of `xparse` support?

**Comment by edemaine** [comment, marked as answer]
I can confirm that KaTeX doesn't support `xparse`. I'm not familiar with xparse, but I imagine we could add it in, if you're interested in working on KaTeX.

In general I feel like KaTeX has a pretty flexible and powerful macro system (with some things still missing though), but I'm also not very familiar with MathJax's so I'm not sure how they compare.

If you have a specific example of a macro you'd like to implement, I'd be happy to think about how to best express it in KaTeX.
